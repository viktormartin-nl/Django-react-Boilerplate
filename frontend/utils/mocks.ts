import { Tip } from "../contexts/LandingPageContext";

export const tips: Tip[] = [
  { label: "Plan your application", key: "1" },
  { label: "Don't repeat yourself", key: "2" },
  { label: "Keep it simple, stupid", key: "3" },
  { label: "Control change", key: "4" },
  { label: "Document what you're doing", key: "5" },
  { label: "Test what you're writing", key: "6" }
];
