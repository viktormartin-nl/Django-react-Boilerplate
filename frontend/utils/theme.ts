import baseStyled, { ThemedStyledInterface } from "styled-components";

export const theme = {
  fontFamily: "font-family: 'Nunito Sans', sans-serif;",
  color: {
    white1: "#fff",
    white2: "rgba(255, 255, 255, 0.5)",
    white3: "#f7f7f7",
    blue1: "#0b3e71",
    blue2: "#1899d6",
    blue3: "#1cb0f6",
    blue4: "#235390",
    green1: "#78c800",
    green2: "#58a700",
    grey1: "#e5e5e5",
    grey2: "#4b4b4b"
  }
};

export type Theme = typeof theme;
export const styled = baseStyled as ThemedStyledInterface<Theme>;
