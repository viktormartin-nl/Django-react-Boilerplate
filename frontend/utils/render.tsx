import React from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "./theme";
import { LandingPageProvider } from "../contexts/LandingPageContext";

export const withTheme = (children: React.ReactElement) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const withContextProvider = (children: React.ReactElement) => {
  return <LandingPageProvider>{children}</LandingPageProvider>;
};
