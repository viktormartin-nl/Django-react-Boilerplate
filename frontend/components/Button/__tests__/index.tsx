import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { Button, ButtonType } from "..";
import { withTheme } from "../../../utils/render";
import { fireEvent } from "@testing-library/react";
import { theme } from "../../../utils/theme";
import { render } from "@testing-library/react";

describe("<Button /> test case", () => {
  test("test ids are in the document", () => {
    const { getByTestId } = render(withTheme(<Button label="Click me" />));

    expect(getByTestId("buttonWrapper")).toBeInTheDocument();
    expect(getByTestId("styledButton")).toBeInTheDocument();
    expect(getByTestId("label")).toBeInTheDocument();
  });

  test("onClick prop is called", () => {
    const onClick = jest.fn();
    const { getByTestId } = render(
      withTheme(<Button label="Click me" onClick={onClick} />)
    );

    fireEvent.click(getByTestId("styledButton"), { button: 1 });
    expect(onClick).toHaveBeenCalled();
  });

  test("implicitly renders the primary style", () => {
    const { getByTestId } = render(withTheme(<Button label="Click me" />));

    const styledButton = getByTestId("styledButton");

    expect(styledButton).toHaveStyle(
      `background-color: ${theme.color.green1};
      color: ${theme.color.white1};`
    );
    expect(styledButton).toMatchSnapshot();
  });

  test("explicitly renders the primary style", () => {
    const { getByTestId } = render(
      withTheme(<Button label="Click me" type={ButtonType.primary} />)
    );

    const styledButton = getByTestId("styledButton");

    expect(styledButton).toHaveStyle(
      `background-color: ${theme.color.green1};
      color: ${theme.color.white1};`
    );
    expect(styledButton).toMatchSnapshot();
  });

  test("renders secondary style", () => {
    const { getByTestId } = render(
      withTheme(<Button label="Click me" type={ButtonType.secondary} />)
    );

    const styledButton = getByTestId("styledButton");

    expect(styledButton).toHaveStyle(
      `background-color: ${theme.color.white1};
      color: ${theme.color.blue1};`
    );
    expect(styledButton).toMatchSnapshot();
  });

  test("renders tertiary style", () => {
    const { getByTestId } = render(
      withTheme(<Button label="Click me" type={ButtonType.tertiary} />)
    );

    const styledButton = getByTestId("styledButton");

    expect(styledButton).toHaveStyle(
      `background-color: ${theme.color.blue3};
      color: ${theme.color.white1};`
    );
    expect(styledButton).toMatchSnapshot();
  });

  test("renders passed label prop", () => {
    const { getByTestId } = render(withTheme(<Button label="Click me" />));

    expect(getByTestId("label")).toHaveTextContent("Click me");
  });
});
