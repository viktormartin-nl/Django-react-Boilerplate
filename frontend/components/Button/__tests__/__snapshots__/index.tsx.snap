// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> test case explicitly renders the primary style 1`] = `
.c0 {
  position: relative;
  border-color: transparent;
  outline: none;
  border-top-left-radius: 16px;
  border-top-right-radius: 16px;
  border-bottom-right-radius: 18px;
  border-bottom-left-radius: 18px;
  border-width: 0 0 4px;
  padding: 13px 16px;
  background-color: #78c800;
  color: #fff;
  cursor: pointer;
}

.c0:after {
  position: absolute;
  background-color: #58a700;
  border-color: transparent;
  border-width: 0 0 4px;
  border-radius: 16px;
  bottom: -8px;
  content: "";
  left: 0;
  right: 0;
  z-index: -1;
  top: 0;
}

.c0:focus {
  outline: none;
}

.c1 {
  position: relative;
  text-transform: uppercase;
  font-weight: bold;
  top: 1px;
}

<button
  class="c0"
  data-testid="styledButton"
>
  <span
    class="c1"
    data-testid="label"
  >
    Click me
  </span>
</button>
`;

exports[`<Button /> test case implicitly renders the primary style 1`] = `
.c0 {
  position: relative;
  border-color: transparent;
  outline: none;
  border-top-left-radius: 16px;
  border-top-right-radius: 16px;
  border-bottom-right-radius: 18px;
  border-bottom-left-radius: 18px;
  border-width: 0 0 4px;
  padding: 13px 16px;
  background-color: #78c800;
  color: #fff;
  cursor: pointer;
}

.c0:after {
  position: absolute;
  background-color: #58a700;
  border-color: transparent;
  border-width: 0 0 4px;
  border-radius: 16px;
  bottom: -8px;
  content: "";
  left: 0;
  right: 0;
  z-index: -1;
  top: 0;
}

.c0:focus {
  outline: none;
}

.c1 {
  position: relative;
  text-transform: uppercase;
  font-weight: bold;
  top: 1px;
}

<button
  class="c0"
  data-testid="styledButton"
>
  <span
    class="c1"
    data-testid="label"
  >
    Click me
  </span>
</button>
`;

exports[`<Button /> test case renders secondary style 1`] = `
.c0 {
  position: relative;
  border-color: transparent;
  outline: none;
  border-top-left-radius: 16px;
  border-top-right-radius: 16px;
  border-bottom-right-radius: 18px;
  border-bottom-left-radius: 18px;
  border-width: 0 0 4px;
  padding: 13px 16px;
  background-color: #fff;
  color: #0b3e71;
  cursor: pointer;
}

.c0:after {
  position: absolute;
  background-color: rgba(255,255,255,0.5) border-color:transparent;
  border-width: 0 0 4px;
  border-radius: 16px;
  bottom: -8px;
  content: "";
  left: 0;
  right: 0;
  z-index: -1;
  top: 0;
}

.c0:focus {
  outline: none;
}

.c1 {
  position: relative;
  text-transform: uppercase;
  font-weight: bold;
  top: 1px;
}

<button
  class="c0"
  data-testid="styledButton"
>
  <span
    class="c1"
    data-testid="label"
  >
    Click me
  </span>
</button>
`;

exports[`<Button /> test case renders tertiary style 1`] = `
.c0 {
  position: relative;
  border-color: transparent;
  outline: none;
  border-top-left-radius: 16px;
  border-top-right-radius: 16px;
  border-bottom-right-radius: 18px;
  border-bottom-left-radius: 18px;
  border-width: 0 0 4px;
  padding: 13px 16px;
  background-color: #1cb0f6;
  color: #fff;
  cursor: pointer;
}

.c0:after {
  position: absolute;
  background-color: #1899d6;
  border-color: transparent;
  border-width: 0 0 4px;
  border-radius: 16px;
  bottom: -8px;
  content: "";
  left: 0;
  right: 0;
  z-index: -1;
  top: 0;
}

.c0:focus {
  outline: none;
}

.c1 {
  position: relative;
  text-transform: uppercase;
  font-weight: bold;
  top: 1px;
}

<button
  class="c0"
  data-testid="styledButton"
>
  <span
    class="c1"
    data-testid="label"
  >
    Click me
  </span>
</button>
`;
